flow/energy_models/power_demand.py

```diff
@@ -0,0 +1,54 @@
+import math
+import random
+import statistics
+import numpy as np
+from scipy.interpolate import interp1d
+from collections import namedtuple
+from flow.core.kernel.base_energy import BaseEnergyModel
+from abc import ABCMeta, abstractmethod
+
+class PowerDemandModel(BaseEnergyModel, metaclass=ABCMeta):
+    """Calculate power consumption of a vehicle.
+    Assumes vehicle is an average sized vehicle.
+    The power calculated here is the lower bound of the 
+    actual power consumed by the vehicle
+    """
+
+    def __init__(self,kernel):
+        self.k = kernel
+        self.g = 9.81
+        self.rho_air = 1.225
+        self.mass = 1200
+        self.rolling_res_coeff = 0.005
+        self.aerodynamic_drag_coeff = 0.3
+        self.cross_area = 2.6
+        self.gamma = 1
```

Comment:
some of these numbers are specific to prius (move to PDMElectric), and some need to be modified for PDMCombustionEngine